### DISCUSSION TITLE
[@types/d3] Add flatRollup(values, reduce, ...keys)

### DISCUSSION THREAD

**Discussion by jsw** [DefinitelyTyped/DefinitelyTyped#62830] [Issues with a @types package]
Please add type definition for this function

https://github.com/d3/d3-array/blob/04f3423768daeda9fc6109687b5c1aa31000c278/src/group.js#L23-L25

```
export function flatRollup(values, reduce, ...keys) {
  return flatten(rollups(values, reduce, ...keys), keys);
}
```

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
@jsw `flatRollup` exists in `@types/d3-array`. Check your version.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/86bb5fee65560e97d0900177d054712400511dee/types/d3-array/index.d.ts#L712-L753

**Reply by jsw** [reply]
Hi @Semigradsky I'm looking for the specific signature in my original post, where it takes an array of keys.

**Reply by Semigradsky** [reply]
It doesn't accept an array of keys. It accepts separate keys as aguments.
```ts
flatRollup(values, reduce, [key1, key2, key3]) // wrong
flatRollup(values, reduce, key1, key2, key3) // correct
```

**Reply by jsw** [reply]
Sorry, my terminology was misleading. I'll try again. The `...` "rest parameter" allows a function to accept an indefinite number of arguments *as an array*. 

```
> function f(...x) { return x; }
undefined
> f(1)
[ 1 ]
> f(1,2,3,4,5)
[ 1, 2, 3, 4, 5 ]
```

@types/d3 only allows up to 3 keys. 

```
flatRollup(values, reduce, key1, key2, key3); // ok
flatRollup(values, reduce, key1, key2, key3, key4); // not ok
```

The function in my original post accepts a rest parameter. Is it possible to update the types library to do the same?

**Reply by Semigradsky** [reply]
It makes sense!

Need to add variants with more parameters and variant with rest parameter. For example, as for `flow` in `lodash`:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bab43e26fbd2faf2dd611c7ff99ed30cd6f98f78/types/lodash/common/util.d.ts#L169-L204
